src/main/java/org/aksw/simba/lemming/metrics/single/nodetriangles/ayz/ListingAyzMetric.java

```diff
@@ -34,32 +35,38 @@ public ListingAyzMetric() {
 
     @Override
     public double apply(ColouredGraph graph) {
-        double delta = 3.0;
-        double threshold = Math.pow(graph.getGraph().getNumberOfEdges(), (delta - 1) / (delta + 1));
-        IntSet highDegreeVertices = new DefaultIntSet(Constants.DEFAULT_SIZE); 
-        		//IntSets.from();
+
+        double r = 2.376;
+        double threshold = Math.pow(graph.getGraph().getNumberOfEdges(), (r - 1) / (r + 1));
+        IntSet highDegreeVertices = new DefaultIntSet(Constants.DEFAULT_SIZE);
+        IntSet lowDegreeVertices = new DefaultIntSet(Constants.DEFAULT_SIZE);
+
+        //divide vertices into lowDegreeVertices and highDegreeVertices
         for (int vertex:graph.getVertices()) {
             if (graph.getGraph().getVertexDegree(vertex) > threshold) {
                 highDegreeVertices.add(vertex);
+            }else{
+                lowDegreeVertices.add(vertex);
             }
         }
 
+        //NodeIterator performed on lowDegreeVertices, matrix multiplication induced subgraph of highDegreeVertices
         double numberOfTriangles = 0;
-        numberOfTriangles += countTrianglesViaNodeIterator(graph, highDegreeVertices);
+        numberOfTriangles += countTrianglesViaNodeIterator(graph, lowDegreeVertices);
         numberOfTriangles += countTrianglesViaMatrixMultiplication(graph, highDegreeVertices);
 
         return numberOfTriangles;
     }
 
 
-    private double countTrianglesViaNodeIterator(ColouredGraph graph, IntSet highDegreeVertices) {
+    private double countTrianglesViaNodeIterator(ColouredGraph graph, IntSet vertices) {
```

Comment:
Please add a (short) javadoc comment that explains the method.

Reply:
Here I made a mistake, NodeIteratorMetric is performed on V/highDegreeVertices. I've changed it back.